### ISSUE TITLE
'Exception' with message 'Could not find the requested document.' in render_default.plugin.php:32

### ISSUE THREAD

**Issue by Pilskalns** [kasparsd/orbiter#2]
Wanted to quick try out this tool, but got into setup issues:
There is no folder `cache`. Auto-create-it-feature would be cool. I will create pull for it, if get over next error:

> Fatal error:  Uncaught exception 'Exception' with message 'Could not find the requested document.' in C:\nginx\html\orbiter\plugins\render_default\render_default.plugin.php:32
> Stack trace:
> # 0 [internal function]: render_default->render_item('', Array)
> # 1 C:\nginx\html\orbiter\index.php(239): call_user_func(Array, '', Array)
> # 2 C:\nginx\html\orbiter\index.php(171): orbiter::filter('render', '')
> # 3 C:\nginx\html\orbiter\index.php(28): orbiter->render()
> # 4 C:\nginx\html\orbiter\index.php(4): orbiter->run()
> # 5 {main}
> 
>  thrown in C:\nginx\html\orbiter\plugins\render_default\render_default.plugin.php on line 32`

**Comment by kasparsd** [comment]
@Pilskalns Great suggestion, thanks! Did you manage to get it working? Here are the [source files for viaorbiter.com](https://github.com/kasparsd/viaorbiter.com) which could serve you as an example.

**Comment by Pilskalns** [comment]
@kasparsd yes, I get over cache folder by just creating it, but it is not why I created this issue: "fatal error" stays. It is out-of-the-box unzipped copy of Orbiter. Some time later I will check deeper why this error is thrown.

**Comment by Pilskalns** [comment]
Looks I have get a bit further and found solution to this issue. The problem was caused by `build_index()` function inside `index.php`. When generating `cahce/index.json` all key values inside array got _double-secaped_ url's when url should be without directory. Like "\\/index" instead of "index" when etc...

I just added 2 letters of `\\` in line 99 and it started working (if not counting all those debug outputs I had to try) from

``` php
trim( $uri_info['dirname'], '/' ),
```

to

``` php
trim( $uri_info['dirname'], '/\\' ),
```

I believe this Could be OS related, but it happens for me on both - as localhost, as well as windows & unix system OS shared host, so it is bit mystical up for now... Also, I will create pull request for this issue.
